### PR TITLE
Update to clap 4.

### PR DESCRIPTION
**Pull request by qwandor**
Most other crates (including `cxx`) use clap 4 by now, and bindgen using clap 3 results in multiple versions needing to be maintained in projects which vendor dependencies.

### CODE REVIEW

**Comment by qwandor** [comment]
Hi, sorry, I've added a description. The context in my case is that I'm working on Android, where we vendor all dependencies into our tree, and require that there is only a single version of each crate. We would like to update to clap 4, as the latest version of a number of other crates that we use depend on it, and would also like to use some features from it in our own code. However, we also use bindgen, which currently still depends on clap 3, which is blocking the update.

I could patch bindgen downstream in our vendored version, but given the relatively large number of changes that's going to make maintenance a real pain, so I'd much rather get this update in upstream if possible.

**Comment by pvdrz** [comment]
Oh ok yeah, that sounds like a hassle.

I'm not sure but maybe we can raise the MSRV for `bindgen-cli` only which is the crate that depends on `clap`. Although you shouldn't have to vendor `bindgen-cli` if you're not using it. I guess that's a consequence of the repo being a workspace instead of a bunch of standalone crates.

I'll try to figure out something so we can merge this ASAP.

**Comment by qwandor** [comment]
Thanks!

We are using `bindgen-cli`, as part of our build system I think. We generally import crates from crates.io rather than git, so being a workspace vs. standalone crates doesn't matter too much.

**Comment by pvdrz** [comment]
> Thanks!
> 
> We are using `bindgen-cli`, as part of our build system I think. We generally import crates from crates.io rather than git, so being a workspace vs. standalone crates doesn't matter too much.

huh, interesting. So you actually need to pull the `bindgen-cli` code.

**Comment by qwandor** [comment]
Right, you can see it [here](https://cs.android.com/android/platform/superproject/+/master:external/rust/crates/bindgen-cli/) if you're curious.

**Comment by qwandor** [comment]
> umbrella The latest upstream changes (presumably [5875949](https://github.com/rust-lang/rust-bindgen/commit/5875949e8e598fa9b0e3a018b3547815e2654fc8)) made this pull request unmergeable. Please resolve the merge conflicts.

Done.

File: bindgen-cli/options.rs [hunk 1]
```diff
@@ -289,112 +306,128 @@ where
...
                 .help(
                     "Disable support for native Rust unions.",
-                ),
+                )
+                .action(ArgAction::SetTrue),
             Arg::new("disable-header-comment")
                 .long("disable-header-comment")
                 .help("Suppress insertion of bindgen's version identifier into generated bindings.")
-                .multiple_occurrences(true),
+                .action(ArgAction::SetTrue),
```
Comment (amanjeev) [inline bindgen-cli/options.rs:354]: Ah! this is neat. It seems we were using `multiple_occurrences` incorrectly. Thank you! :heart:

**Review by amanjeev — APPROVED** [review]
Thank you for working on this and opening this PR.

**Comment by amanjeev** [comment]
r+

**Pull request merged**